src/utils/async.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Maps over an AsyncIterator
+ */
+export function mapAsyncIterator<T, V>(
+  iter: AsyncIterator<T>,
+  f: (t: T) => V
+): AsyncIterator<V> {
+  return {
+    next: () =>
+      iter.next().then(({ done, value }) => ({
+        done,
+        value: f(value)
+      }))
+  };
+}
+
+/**
+ * Maps over an AsyncIterable
+ */
+export function mapAsyncIterable<T, V>(
+  source: AsyncIterable<T>,
+  f: (t: T) => V
+): AsyncIterable<V> {
+  const iter = source[Symbol.asyncIterator]();
+  const iterMapped = mapAsyncIterator(iter, f);
+  return {
+    [Symbol.asyncIterator]: () => iterMapped
+  };
+}
+
+export async function asyncIteratorToArray<T>(
+  iter: AsyncIterator<T>
+): Promise<ReadonlyArray<T>> {
+  const acc = Array<T>();
+
+  while (true) {
```

Comment:
Is it possible to use hasMoreResults or similar?

Reply:
Nevermind, I was looking for another way to iterate over next elements of the iterator, but it seems to be the only way.